dev/Pager/PrototypePager/PrototypePager.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+
+// The Templated Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234235
+
+namespace MUXControlsTestApp
+{
+    public sealed partial class PrototypePager : Control
+    {
+        public enum PagerDisplayModes { Auto, ComboBox, NumberBox, NumberPanel, }
+        public enum ButtonVisibilityMode { Auto, AlwaysVisible, HiddenOnEdge, None, }
+
+        private Button FirstPageButton, PreviousPageButton, NextPageButton, LastPageButton;
+
+        public event TypedEventHandler<PrototypePager, PageChangedEventArgs> PageChanged;
+        
+        public PrototypePager()
+        {
+            this.DefaultStyleKey = typeof(PrototypePager);
+            this.Loaded += (s, args) => { PagerTemplateSettings = new PagerTemplateSettings(this); };
+
+        }
+        protected override void OnApplyTemplate()
+        {
+            // Grab UIElements for later
+            FirstPageButton = GetTemplateChild<Button>("FirstPageButton");
+            PreviousPageButton = GetTemplateChild<Button>("PreviousPageButton");
+            NextPageButton = GetTemplateChild<Button>("NextPageButton");
+            LastPageButton = GetTemplateChild<Button>("LastPageButton");
+            FirstPageButtonTestHook = FirstPageButton;
+            PreviousPageButtonTestHook = PreviousPageButton;
+            NextPageButtonTestHook = NextPageButton;
+            LastPageButtonTestHook = LastPageButton;
+            NumberBoxDisplayTestHook = GetTemplateChild<NumberBox>("NumberBoxDisplay");
+            ComboBoxDisplayTestHook = GetTemplateChild<ComboBox>("ComboBoxDisplay");
+
+            // Attach Callbacks for property changes
+            RegisterPropertyChangedCallback(SelectedIndexProperty, DisablePageButtonsOnEdge);
+            RegisterPropertyChangedCallback(SelectedIndexProperty, (s, e) => { PageChanged?.Invoke(this, new PageChangedEventArgs(SelectedIndex - 1)); });
+
+            // Attach click events
+            FirstPageButton.Click += (s, e) => { SelectedIndex = 1; };
```

Comment:
>FirstPageButton [](start = 12, length = 15)

You need to check to make sure that this is not null, If someone has retemplated the control to a different template it might not have a FirstPageButton and thus the GetTemplateChild will return null.

Reply:
And now I see you are throwing on null below.  We need to allow people to retemplate and continue to function event if the button doesn't exist.

---
In reply to: [463287960](https://github.com/microsoft/microsoft-ui-xaml/pull/2977#discussion_r463287960) [](ancestors = 463287960)

Reply:
Acknowledged! I have refactored the code for this

Reply:
Nit: in c# you have the ? operator so you could replace this with `FirstPageButton?.Click += (s, e) => { SelectedIndex = 1; }

---
In reply to: [463890550](https://github.com/microsoft/microsoft-ui-xaml/pull/2977#discussion_r463890550) [](ancestors = 463890550)

Reply:
Actually, that line will result in an error. My first thought was to add that operator in as well but didn't work out unfortunately.

Reply:
Whats the error?

Reply:
The event 'ButtonBase.Click' can only appear on the left hand side of += or -=.

Reply:
I came across this post which talks a bit more about this error. https://stackoverflow.com/questions/49688764/subscribe-to-events-of-a-nullable-class